core/src/main/java/org/jruby/util/IOOutputStream.java

```diff
@@ -67,9 +67,22 @@
     public IOOutputStream(final IRubyObject io, Encoding encoding, boolean checkAppend, boolean verifyCanWrite) {
         this.io = io;
         this.runtime = io.getRuntime();
-        this.realIO = ( io instanceof RubyIO && !((RubyIO) io).isClosed() &&
-                ((RubyIO) io).isBuiltin("write") ) ?
-                    ((RubyIO) io) : null;
+
+        // If we can get a real IO from the object, we can do fast writes
+        RubyIO realIO = null;
```

Comment:
Not saying do this but this section could be RubyIO realIO = determineRealIO(...); since the code block only calcs a single thing.

PS- I don't care one way or other :)

Reply:
I actually did that at one point but removed it to keep diff smaller. I think it is a fine change now, though.